onnxruntime/core/framework/session_state.cc

```diff
@@ -308,4 +456,38 @@ const NodeIndexInfo& SessionState::GetNodeIndexInfo() const {
   ORT_ENFORCE(node_index_info_, "SetGraphAndCreateKernels must be called prior to GetExecutionInfo.");
   return *node_index_info_;
 }
+
+void SessionState::UpdateToBeExecutedNodes(const std::vector<int>& fetch_mlvalue_idxs) {
+  if (to_be_executed_nodes_.find(fetch_mlvalue_idxs) != to_be_executed_nodes_.end())
+    return;
```

Comment:
Does this 'find' do a lookup involving every entry in fetch_ml_value_idxs when attempting to match each key in to_be_executed_nodes_? Naively I would expect it to involve a comparison of 2 vectors every time.